src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -1052,14 +1056,53 @@ julia> project(P, [1, 2]) |> constraints_list
 ```
 """
 function project(P::AbstractPolyhedron{N}, block::AbstractVector{Int}) where {N}
-    if constrained_dimensions(P) ⊆ block
-        clist = [HalfSpace(c.a[block], c.b) for c in constraints_list(P)]
-    else
+    general_case = false
+
+    # cheap case
+    clist = nothing  # allocate later
+    @inbounds for c in constraints(P)
+        subset_status = 0  # initially 0, set to 1 if subset and -1 if disjoint
+        for i in constrained_dimensions(c)
+            if i in block
+                # case 1: is a subset
+                if subset_status == -1
+                    general_case = true
+                    break
+                elseif subset_status == 0
+                    subset_status = 1
+                end
+            else
+                # case 1: is disjoint
+                if subset_status == 1
+                    general_case = true
+                    break
+                elseif subset_status == 0
+                    subset_status = -1
+                end
+            end
+        end
+        if subset_status == 1
+            # simple projection of half-space
+            hs = HalfSpace(c.a[block], c.b)
+            if clist == nothing
+                clist = [hs]  # get the right type of the constraints
+            else
+                push!(clist, hs)
+            end
+        end
+    end
+
+    # general case
+    if general_case
```

Comment:
can you outsource the general case to a separate function `_project_polyhedron` or sth?

Reply:
done